### PR TITLE
Documentation and Testing Updates

### PR DESCRIPTION
**Pull request by spaulaus**
## Description
This PR includes updates to documentation and testing. The documentation updates revolve around 
making this project more contributor friendly. It resolves the three missing components seen on the 
[community](https://github.com/spaulaus/paass-laughing-conqueror/community) page. 

The testing updates allow the unit tests to be run using `ctest` or `make test`, which improves the ease 
of testing the software. We will no longer need to run the tests via bash scripts or even install them should we choose not to. 

## Related Issues
* #95 - Adds build status to README. 
* #96 - CMake test support
* #98 - Updates the Community Documentation

## Motivation and Context
We are looking to build a community of support, which starts with making it easy for them to get going with the project.

## How Has This Been Tested?
Tested using [Travis-CI](https://github.com/spaulaus/TravisCI-Tests). All of the updates that are included
here have been tested on that repository first. I have also compiled and run all unittests. 
**Note:** : The links to the Code of Conduct and the Contributing document will not be live until this is pushed up to master. 

## Screenshots (if appropriate):
None 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+[Code of Conduct](https://github.com/spaulaus/paass-laughing-conqueror/blob/master/.github/CODE_OF_CONDUCT.md).
+
+# Where We Need Help
+* Unit Testing
+* Documentation
+  * Wiki (formatting / content updates)
+  * How To Articles
+  * Code Documentation
+
+# Pull Request Process
```
Comment (ksmith0) [inline .github/CONTRIBUTING.md]: Thanks for posting this. I seem to always forget what we had decided upon.

File: README.md [hunk 2]
```diff
@@ -25,24 +32,24 @@ that can be extended beyond just a single group or digitizer.
...
 [original project](https://github.com/pixie16/paass).   
 
-# Dependencies
-## For all components 
+## Dependencies
+### For all components 
 * ncurses
 * gcc v4.4.6+
 * cmake v2.8.8+
 * [ROOT](https://root.cern.ch)
```
Comment (ksmith0) [inline README.md]: Do we have a version requirement for ROOT? Are we still ROOT 5 compatible?

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (spaulaus) [reply .github/CONTRIBUTING.md]: This is more for external contributors. I need to flesh this out more generally to encompass how I typically operate.

File: README.md (hunk 2 above)
Comment (spaulaus) [reply README.md]: We are still ROOT5 compatible. That's the only Version we can easily test with in the CI server.

Comment (ksmith0) [reply README.md]: Because it is in the package manager for the current travis os?

Comment (spaulaus) [reply README.md]: That is correct.

**Comment by spaulaus** [comment]
This issue was moved to spaulaus/paass#99

**Pull request merged**